packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -222,7 +222,12 @@ function s(
                     // undefined is for root components, but root components cannot accept slotted content
                     setVMBeingRendered(slotset.owner!);
                     try {
-                        ArrayPush.call(newChildren, vnode.factory(data.slotData, data.key));
+                        // The factory function is a template snippet from the slot set owner's template,
+                        // hence switch over to the slot set owner's template reactive observer
+                        const { tro } = slotset.owner!;
+                        tro.observe(() => {
```

Comment:
Ok, but now what happen with the other part? if you have a local value from the child? At the end of the day, the iteration if on the child... and whatever the parent is passing down is passed down... I don't fully understand the problem here!